web_poet/page_inputs/http.py

```diff
@@ -36,12 +37,16 @@ class HttpRequestBody(bytes):
 class HttpResponseBody(bytes):
     """A container for holding the raw HTTP response body in bytes format."""
 
+    def bom_encoding(self) -> Optional[str]:
+        """Returns the encoding from the byte order mark if present."""
+        return read_bom(self)[0]
+
     def declared_encoding(self) -> Optional[str]:
         """Return the encoding specified in meta tags in the html body,
         or ``None`` if no suitable encoding was found"""
         return html_body_declared_encoding(self)
 
-    def json(self):
+    def json(self) -> Optional[str]:
```

Comment:
This type annotation is wrong, any object can be returned, as it's result of json.loads

Reply:
Nice catch! I was on a copy-pasting spree on the past commit. 🤦 

Although I'm curious as to why mypy wasn't able to detect this one as the tests were comparing it to a `dict`.